### PR TITLE
Movies & Screenings Page

### PR DESCRIPTION
**Pull request by WawrzenskiPatryk**
### Worth to mention changes

- New / changed components:
  - Base components: `BaseHeading`, `BaseInput`, `BaseSelect`, `BaseTag`
  - Several components for screenings page and panels
  - Several components for movies page and panels
  - `LandingPanel` styling change
  - `MoviesListPage` deleted - logic moved to `MoviesPage` + related routing change
- `HomePage` expansion
- `vue-datepicker` package implementation
- Movies filtering logic by title and genre -  `src/store` and components
- Screenings data fetching, storing, formatting, displaying and filtering logic - `src/services`,  `src/store` and components
- Service functions for managing dates in `src/services/dates.js`
- google font import: Eczar 400, 500, 700
- Other changes to already existing files

### CODE REVIEW

File: src/components/base/BaseHeading.vue [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<script>
+import { defineComponent } from 'vue';
+
+export default defineComponent({
+  props: {
+    headingSize: {
+      type: String,
+      default: 'large',
```
Comment (flipnow) [inline src/components/base/BaseHeading.vue]: Since the prop is required I think it doesn't make sense to have a default value. Also valid would make it not required and have a default value instead :)

File: src/components/movies/MovieDetail.vue [hunk 2]
```diff
@@ -1,26 +1,136 @@
...
+      else backgroundPosition = '50% 50%';
+
+      return {
+        'background-image': `url(${this.movie.poster_url})`,
+        'background-position': backgroundPosition,
+      };
+    },
+    movieReleaseYear() {
+      const indexOfFirstHyphen = this.movie.release_date.indexOf('-');
+      return this.movie.release_date.slice(0, indexOfFirstHyphen);
```
Comment (flipnow) [inline src/components/movies/MovieDetail.vue]: I think a bit less hacky solution could be to simply parse your `release_date` to `Date` object and then retrieve a year

```suggestion
      const date = new Date(this.move.release_date);
      return date.getFullYear();
```

File: src/components/movies/MovieDetail.vue [hunk 3]
```diff
@@ -1,26 +1,136 @@
...
+    width: 34%;
+    @include screen-min-large {
+      width: 45%;
+    }
+  }
+
+  &__title {
+    margin: 0;
+
+    font-family: 'Eczar', serif;
```
Comment (flipnow) [inline src/components/movies/MovieDetail.vue]: It seems like you want all titles to have Eczar font. It would be a good idea to make your base heading component a bit more flexible and reuse it for all headings. It will make it easier for you  to keep your styling consistent and so you don't have to remember about setting it everywhere. Additionally, if Roboto is used for all the other text, you could specify it in global styles.

File: src/components/movies/MovieDetail.vue [hunk 4]
```diff
@@ -1,26 +1,136 @@
...
     movie: {
       type: Object,
       required: true,
     },
   },
+  computed: {
+    imageStyles() {
+      let backgroundPosition;
+
+      if (this.movie.id === 1) backgroundPosition = '50% 30%';
```
Comment (flipnow) [inline src/components/movies/MovieDetail.vue]: Why background position needs to be hardcoded?

File: src/components/screenings/ScreeningsListSeance.vue [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    seanceData: {
+      type: Object,
+      required: true,
+    },
+  },
+  computed: {
+    seanceTime() {
+      const datetime = this.seanceData.datetime;
+      const timePoint = datetime.indexOf('T') + 1;
+      return datetime.slice(timePoint).slice(0, 5);
```
Comment (flipnow) [inline src/components/screenings/ScreeningsListSeance.vue]: Nice alternative that's supported by browsers would be to use `toLocaleTimeString`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleTimeString

```suggestion
      const date = new Date(this.seanceData.datetime));
      return date.toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' }));
```

File: src/pages/LandingPanel.vue [hunk 6]
```diff
@@ -10,10 +10,8 @@ export default defineComponent({});
...
 
-<style lang="scss">
-.landing-panel {
-  display: flex;
-  flex-direction: column;
-  gap: 6.4rem;
-}
+<style lang="scss" scoped>
+  .landing-panel {
+    min-height: calc(100vh + 1px);
```
Comment (flipnow) [inline src/pages/LandingPanel.vue]: Why was this needed?

Comment (flipnow) [reply src/pages/LandingPanel.vue]: What I mean is why `+ 1px`

File: src/services/dates.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    displayDate: getDisplayDateFormat(year, actualMonth, day),
+    weekday: getWeekdayName(weekdayIndex),
+    shortWeekday: getWeekdayName(weekdayIndex).slice(0, 3),
+  };
+}
+
+function getFollowingDays(todaysYear, todaysMonth, todaysDay) {
+  const followingDays = [];
+  for (let i = 1; i <= 5; i++) {
+    const followingDayInstance = new Date(todaysYear, todaysMonth, todaysDay + i);
```
Comment (flipnow) [inline src/services/dates.js:52]: What will happen if todaysDay is 30?

**Review by flipnow — APPROVED** [review]
Nice one, lots of improvements 👍 I've left a couple of comments and questions

File: src/components/movies/MovieDetail.vue (hunk 4 above)
Comment (WawrzenskiPatryk) [reply src/components/movies/MovieDetail.vue]: Images used for movies description are not anyhow standardised, like having same dimensions or focus point. So I did such thing just to please myself with (which is only my opinion) specific positioning of each image. I guess we can't really mathematically nor logically set positioning without standardised images when it's purely based on it's content. 

Otherwise I could just skip this and center all the background images, but in some cases that resulted with having our posters' blank areas in the center of their container.

File: src/pages/LandingPanel.vue (hunk 6 above)
Comment (WawrzenskiPatryk) [reply src/pages/LandingPanel.vue]: That's a leftover code and I'm sorry for leaving it here.
As far as I remember the purpose was to have a quick fix of screen-jumping behaviour while data to be displayed on the screen was re-fetching. Loading element shown before displaying the fetched content was not big enough to make screen scrollable. I didn't want to make every screen on `overscroll: visible` or something like that so that was the idea for a moment.
Anyway, I figured out how to deal with that jumpy behaviour programmatically with correct `v-if` statement in the template and the snipped you asked about is no longer needed. Will be deleted.

File: src/services/dates.js (hunk 7 above)
Comment (WawrzenskiPatryk) [reply src/services/dates.js:52]: It will go to the next month or to 31st day of the month depending on what is the current month.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#syntax
> If any parameter overflows its defined bounds, it "carries over". For example, if a monthIndex greater than 11 is passed in, those months will cause the year to increment; if a minutes greater than 59 is passed in, hours will increment accordingly, etc. Therefore, new Date(1990, 12, 1) will return January 1st, 1991; new Date(2020, 5, 19, 25, 65) will return 2:05 A.M. June 20th, 2020.

File: src/components/movies/MovieDetail.vue (hunk 3 above)
Comment (WawrzenskiPatryk) [reply src/components/movies/MovieDetail.vue]: I created the base heading component later and just forgot to use it here. Thank you for finding that!

File: src/components/movies/MovieDetail.vue (hunk 4 above)
Comment (flipnow) [reply src/components/movies/MovieDetail.vue]: I would suggest not hardcoding those values if possible. This approach works fine for 10 images but having larger dataset it won't be feasible. I feel like your idea with centring sounds good. Or maybe use `background-size: cover;`

Could you show me a screen snippet of the blank area you mentioned at the end?

File: src/pages/LandingPanel.vue (hunk 6 above)
Comment (flipnow) [reply src/pages/LandingPanel.vue]: Awesome 👍

File: src/services/dates.js (hunk 7 above)
Comment (flipnow) [reply src/services/dates.js:52]: Ok nice 🙂  👍

File: src/components/movies/MovieDetail.vue (hunk 4 above)
Comment (WawrzenskiPatryk) [reply src/components/movies/MovieDetail.vue]: Sure. Here is an example of my positioning:
image

And the centered positioning:
image

This is actually the only "blank space" case to be honest, but others tend to not make sense in their own way, like showing characters legs instead of torso / portrait:
image
image

And I think the designs are clear about the image height has to fit the leftside text height.

Comment (WawrzenskiPatryk) [reply src/components/movies/MovieDetail.vue]: Aaand it already uses `background-size: cover;` 😅

Comment (flipnow) [reply src/components/movies/MovieDetail.vue]: 👍 Does it have to be so wide? If image is vertical maybe it would make sense to make it more narrow? With `background-size: contain`

Comment (flipnow) [reply src/components/movies/MovieDetail.vue]: Had a look at designs and I guess it can't

Comment (flipnow) [reply src/components/movies/MovieDetail.vue]: So one thing that i'd suggest here is to extract this configuration logic to a separate file maybe as an object with movie id as a key.

Comment (WawrzenskiPatryk) [reply src/components/movies/MovieDetail.vue]: I like the idea. To be honest in the real project (not a sandbox one) I would just start a discussion about that and ask designers if we can tweak it all a little bit so it doesn't create such a fancy problem. Anyway, let's go with separate file with the object as you said  🚀 👌
